lib/packwerk/commands/update_deprecations_command.rb

```diff
@@ -0,0 +1,67 @@
+# typed: true
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "benchmark"
+
+require "packwerk/commands/offense_progress_marker"
+require "packwerk/run_context"
+require "packwerk/updating_deprecated_references"
+
+module Packwerk
+  module Commands
+    class UpdateDeprecationsCommand
+      extend T::Sig
+      include OffenseProgressMarker
+      Result = Struct.new(:message, :status)
+
+      def initialize(files:, configuration:, offenses_formatter:, progress_formatter:)
+        @files = files
+        @configuration = configuration
+        @progress_formatter = progress_formatter
+        @offenses_formatter = offenses_formatter
+      end
+
+      sig { returns(Result) }
+      def run
+        @progress_formatter.started(@files)
+
+        all_offenses = T.let([], T.untyped)
+        execution_time = Benchmark.realtime do
+          all_offenses = @files.flat_map do |path|
+            run_context.process_file(file: path).tap do |offenses|
+              mark_progress(offenses: offenses, progress_formatter: @progress_formatter)
+            end
+          end
+
+          updating_deprecated_references.dump_deprecated_references_files
+        end
+
+        @offenses_formatter.show_offenses(all_offenses)
```

Comment:
I think it is an excellent idea to move the offense formatter logic outside of this class. 👏 

If you think about the behavior of the offenses_formatter, it behaves like @out but takes a list of offenses and prints them. This essentially means that we have two ways of how a result message gets emitted from the command class, making it harder to reason about and test.

I would suggest changing the `OffenseFormatter` to return a string rather than printing to the stdout. The string would then be returned as the Result message. Something like this:

```ruby
class OffensesFormatter
      extend T::Sig

      def initialize(style: OutputStyles::Plain)
        @style = style
      end

      sig { params(offenses: T::Array[T.nilable(Offense)]).void }
      def format_offenses(offenses)
        return "No offenses detected 🎉"  if offenses.empty?
       
         <<~CONTENT
           #{offenses_list(offenses)}
           #{offenses_summary(offenses)}
         CONTENT
      end
      
      private
      
      def offenses_list(offenses)
        offenses
          .compact
          .map { |offense| offense.to_s(@style)) }
          .join("\n")
      end

      def offenses_summary(offenses)
        offenses_string = Inflector.default.pluralize("offense", offenses.length)
         "#{offenses.length} #{offenses_string} detected"
      end
  end
```

You would have to change the order when the offenses get printed concerning `@progress_formatter.finished(execution_time)`, but I think that is an acceptable change for what we are getting. Also, keep in mind that my change removes the pretending of an empty new line. The OffensesFormatter shouldn't be concerned about how its result is represented next to other content because that is not its responsibility.